.github/actions/init/action.yaml

```diff
@@ -0,0 +1,133 @@
+#  ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+#  ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+#  ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+#  ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+#  ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+#  ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+#  ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+#  ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+#  ┃ This file is part of the Perspective library, distributed under the terms ┃
+#  ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+#  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+name: "Initialize"
+description: "Initialize the build environment with caching"
+
+inputs:
+    skip_cache:
+        description: "Don't use cache from previous builds"
+
+runs:
+    using: "composite"
+    steps:
+        - name: Clean System
+          uses: AdityaGarg8/remove-unwanted-software@v3
+          if: ${{ runner.os == 'Linux' }}
+          with:
+              remove-android: "true"
+              remove-dotnet: "true"
+              remove-haskell: "true"
+              remove-codeql: "true"
+
+        - name: Install pnpm
+          uses: pnpm/action-setup@v3
+          with:
+              version: 9
+
+        - name: Setup emsdk cache
+          uses: actions/cache@v4
+          id: emsdk-cache
+          if: ${{ inputs.skip_cache == 'false' }}
+          with:
+              path: |
+                  ~/boost_1_82_0/
+                  ~/.emsdk/
+                  ~/.llvm/
+              key: ${{ runner.os }}-emsdk-${{ hashFiles('package.json') }}
+              restore-keys: |
+                  ${{ runner.os }}-emsdk-
+
+        - name: Setup pip cache
+          uses: actions/cache@v4
+          if: ${{ inputs.skip_cache == 'false' }}
+          with:
+              path: |
+                  ~/.cache/pip
+                  ~/py_modules/
+              key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py') }}
```

Comment:
this is gonna be `pyproject.toml` now

Reply:
It's now `requirements.txt` because https://github.com/pypa/pip/issues/12191

Reply:
Implemented in #2659